### PR TITLE
`test_auth.py`: Skip three pytests that fail on Travis CI

### PR DESCRIPTION
**Pull request by cclauss**
Travis CI limits outbound network requests.

https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

### CODE REVIEW

**Comment by mfisher87** [comment]
>Travis CI limits outbound network requests.

Any idea why this just started happening?

We are currently separating out some tests that require EDL in the Travis config into "basic tests" and "behind Earthdata" tests. I feel we should only have one way of skipping Travis - either using Pytest and checking envvars, or using Travis config YML.

If the problem is Travis limiting outbound network requests, I'd expect `pytest icepyx/tests/test_behind_NSIDC_API_login.py` to also fail in Travis after we get past these tests. Should we just move off Travis so we can still run some of these tests (IMO they are integration tests, but that's another conversation ;) ) #547

File: icepyx/tests/test_auth.py [hunk 1]
```diff
@@ -18,15 +18,15 @@ def auth_instance():
...
+@pytest.mark.skipif(os.getenv("TRAVIS"), reason="Skipping this test on Travis CI.")
 def test_get_session(auth_instance):
     print(os.environ)
     assert isinstance(auth_instance.session, requests.sessions.Session)
 
 
 # Test that .s3login_credentials creates a dict with the correct keys
 @pytest.mark.skipif(
-    os.environ.get("TRAVIS") == "true", "Skipping this test on Travis CI."
+    os.environ.get("TRAVIS") == "true", reason="Skipping this test on Travis CI."
```
Comment (mfisher87) [inline icepyx/tests/test_auth.py:29]: Nitpick: Let's use either `os.environ.get` or `os.getenv` in the same way for each mark.

Comment (cclauss) [reply icepyx/tests/test_auth.py:29]: 👍 Once I get it to work, I will do it the same way in all three places.

File: icepyx/tests/test_auth.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+
 import pytest
 import requests
 
 import earthaccess
 
 from icepyx.core.auth import EarthdataAuthMixin
 
+# Are we running on Travis CI?
+TRAVIS = os.environ.get("TRAVIS") == "true"
```
Comment (mfisher87) [inline icepyx/tests/test_auth.py:11]: What do you think of:

```python
skip_if_travis_ci = pytest.mark.skipif(TRAVIS, reason="Skipping this test on Travis CI.")


@skip_if_travis_ci
def test_something:
    ...
```

Comment (cclauss) [reply icepyx/tests/test_auth.py:11]: I have never seen that before.  Does it work?

File: icepyx/tests/test_auth.py (hunk 1 above)
Comment (mfisher87) [reply icepyx/tests/test_auth.py]: Sorry if I'm getting ahead of things here :)

File: icepyx/tests/test_auth.py (hunk 2 above)
Comment (cclauss) [reply icepyx/tests/test_auth.py:11]:
```suggestion
skip_if_travis_ci = pytest.mark.skipif(
    os.environ.get("TRAVIS") == "true", reason="Skipping this test on Travis CI."
)
```

File: icepyx/tests/test_auth.py [hunk 3]
```diff
@@ -16,11 +21,13 @@ def auth_instance():
 
 
 # Test that .session creates a session
+@pytest.mark.skipif(TRAVIS, reason="Skipping this test on Travis CI.")
```
Comment (cclauss) [inline icepyx/tests/test_auth.py:24]:
```suggestion
@skip_if_travis_ci
```

File: icepyx/tests/test_auth.py [hunk 4]
```diff
@@ -16,11 +21,13 @@ def auth_instance():
...
 def test_get_session(auth_instance):
     assert isinstance(auth_instance.session, requests.sessions.Session)
 
 
 # Test that .s3login_credentials creates a dict with the correct keys
+@pytest.mark.skipif(TRAVIS, reason="Skipping this test on Travis CI.")
```
Comment (cclauss) [inline icepyx/tests/test_auth.py:30]:
```suggestion
@skip_if_travis_ci
```

File: icepyx/tests/test_auth.py [hunk 5]
```diff
@@ -30,6 +37,7 @@ def test_get_s3login_credentials(auth_instance):
 
 
 # Test that earthdata_login generates an auth object
+@pytest.mark.skipif(TRAVIS, reason="Skipping this test on Travis CI.")
```
Comment (cclauss) [inline icepyx/tests/test_auth.py:40]:
```suggestion
@skip_if_travis_ci
```

File: icepyx/tests/test_auth.py [hunk 6]
```diff
@@ -1,10 +1,17 @@
+import os
+
 import pytest
 import requests
 
 import earthaccess
 
 from icepyx.core.auth import EarthdataAuthMixin
 
+# Are we running on Travis CI?
```
Comment (cclauss) [inline icepyx/tests/test_auth.py:10]:
```suggestion
```

**Comment by mfisher87** [comment]
> We are currently separating out some tests that require EDL in the Travis config into "basic tests" and "behind Earthdata" tests.
> 
> https://github.com/icesat2py/icepyx/blob/e0e1738737db77714e604035242da32bcd263ad2/.travis.yml#L41-L44C11
> 
> I feel we should only have one way of skipping Travis - either using Pytest and checking envvars, or using Travis config YML.
> 
> If the problem is Travis limiting outbound network requests, I'd expect `pytest icepyx/tests/test_behind_NSIDC_API_login.py` to also fail in Travis after we get past these tests. Should we just move off Travis so we can still run some of these tests (IMO they are integration tests, but that's another conversation ;) ) #547

Hey @cclauss what do you think of this? I'm concerned that when we merge, the "behind Earthdata" tests may fail!

**Comment by cclauss** [comment]
I think we should merge this as is because it solves the current problem and that should get automated tests back to ✅.

It has been a long time since I worked with Travis CI so I cannot guarantee if/how they are limiting outbound network requests.  I would certainly support dropping Travis in favor of GHAs but that might be a bit of work given Conda, etc.

Perhaps `behind_NSIDC` will work just fine on Travis.

**Comment by mfisher87** [comment]
I think I can transition us to GitHub actions pretty quickly, we'll just need the credentials set up! cc @JessicaS11

But we need to wait to merge until the release process is over.

**Comment by mfisher87** [comment]
Since I'm new around here I'm not comfortable reducing test coverage without someone else giving the OK :) @weiji14 thoughts on #558 ?

**Comment by cclauss** [comment]
I am completely lost on this maneuver.

https://github.com/icesat2py/icepyx/pull/559#issuecomment-2287152798
> I think we should merge this as is because it solves the current problem and that should get automated tests back to ✅.

This PR allowed Travis CI to pass.  My three other PRs all pass except for the failing Travis CI tests.  Why can't we get tests back to passing _before_ we break things further?

As I stated above, either get rid of Travis in favor of GitHub Actions or get it to pass so that other PRs are not failing.

**Comment by mfisher87** [comment]
The problem isn't that Travis is failing on all PRs, rather that it's failing on PRs from forks only. Disabling those tests for Travis entirely will reduce the coverage for non-fork PRs, including tests running on `main`. #563 addresses this issue by moving these tests from the phase which runs for PRs to a phase which doesn't run for PRs. Does that seem reasonable to you?

**Pull request closed without merging**